components/post_view/post/index.js

```diff
@@ -75,10 +74,8 @@ function makeMapStateToProps() {
 
         return {
             post,
-            author: getDisplayName(state, post.user_id),
-            reactions: getReactionsForPost(state, post.id),
+            createAriaLabel: (intl) => createAriaLabelForPostId(state, post.id, intl),
```

Comment:
`mapStateToProps` (and the function returned by `makeMapStateToProps` in this case) is called each time anything in the redux state changes, so by returning an arrow function each time it's called, it'll cause the `Post` to re-render any time anything changes, even if the changes are unrelated to this.

To fix this, you should make a selector that only returns a new function when something passed into the function changes. If you look above at the code for `getReplyCount`, you'll see that it passes a bunch of functions into `createSelector` to do just that. The last function passed in calculates some value based on the results of the other parameters. In your case, those first few functions would pull the relevant parts of the redux state and return a new arrow function each time one of those parts changes.

Also, you'll want to make your selector creator work like `makeGetReplyCount` where it you can create an instance of the selector for each instance of the component by calling it inside `makeMapStateToProps`. Since each instance of a selector does its own memoization, you'll want each `Post` to have its own copy to prevent the different copies from overwriting each other.

Reply:
One thing to be careful of though is that you can't just pass the `state` into `createAriaLabelForPostId` in this case since then you'd still want to create a new arrow function any time anything changes. You'll probably want to go back to how it worked before where it takes the relevant fields from the state, but it can do it as part of the selector instead of `mapStateToProps` this time like

```javascript
const createAriaLabel = createSelector(
    (state, post) => post.id,
    (state, post) => getDisplayName(state, post.user_id),
    (state, post) => getReactionsForPost(state, post.id),
    (state, post) =>  get(state, Preferences.CATEGORY_FLAGGED_POST, post.id, null) != null,
    (postId, authorName, reactions, isFlagged) => {
        return (intl) => PostUtils.createAriaLabelForPost(post, author, isFlagged, reactions,intl);
    }
);